clang/lib/Sema/SemaChecking.cpp

```diff
@@ -7870,10 +7884,17 @@ bool Sema::CheckIntelSYCLAllocaBuiltinFunctionCall(unsigned, CallExpr *Call) {
     const TemplateArgumentList &TAL =
         cast<ClassTemplateSpecializationDecl>(Ty->getAsCXXRecordDecl())
             ->getTemplateArgs();
-    return !TAL.get(0).getAsType()->isIntegralType(Ctx);
+    if (!TAL.get(0).getAsType()->isIntegralType(Ctx))
+      return true;
+    llvm::APSInt DefaultSize =
+        getSYCLAllocaDefaultSize(Ctx, cast<VarDecl>(CST->get(1).getAsDecl()));
+    if (DefaultSize < 1)
+      Diag(Loc, diag::warn_intel_sycl_alloca_bad_default_value)
```

Comment:
Should this be an error?   What happens with invalid value allocas?

Reply:
We're warning users against using either of:
```c++
constexpr sycl::specialization_id<size_t> zero;
constexpr sycl::specialization_id<size_t> neg(-1);
```
as a SYCL `private_alloca` (receiving specialization constants as size), as that's UB, even if they set the specialization constants to a valid value before launching the kerrnel.

Reply:
As it's UB, we're just warning

Reply:
@premanandrao LGTY?